cpp/src/timestamps.cpp

```diff
@@ -91,21 +91,20 @@ LogicalColumn to_timestamps(const LogicalColumn& input,
   return ret;
 }
 
-LogicalColumn extract_timestamp_component(const LogicalColumn& input,
-                                          cudf::datetime::datetime_component component)
+LogicalColumn extract_timestamp_component(const LogicalColumn& input, std::string component)
 {
-  if (!cudf::is_timestamp(input.cudf_type())) {
+  auto timestamp_type = std::dynamic_pointer_cast<arrow::TimestampType>(input.arrow_type());
+  if (!timestamp_type) {
     throw std::invalid_argument("extract_timestamp_component() input must be timestamp");
```

Comment:
Hmmm, I guess most of this will already throw in the `std::dynamic_pointer_cast` and this just checks whether the `shared_ptr` points to something.
Should we add a test? (the old error is a bit different, but doesn't really matter.)

Reply:
Good idea, will add a test.